lsudt/lsudt.py

```diff
@@ -474,10 +496,86 @@ def init_argparse() -> argparse.ArgumentParser:
         dest="id_path",
         help="limit output to devices with a parent starting with given idpath",
     )
+    parser.add_argument(
+        "--extract-env",
+        "-x",
+        action="store_true",
+        dest="extract_env",
+        help="print name=value pairs to stdout"
+    )
 
     return parser
 
 
+def generate_and_print_env_strings(label: str):
+    """
+    prints the name=value pairs to stdout
+    """
+
+    strings = ""
+    for key in list(env_pairs.keys()):
+        counter = 0
+        for dev in env_pairs[key]:
+            strings += (f" {label}_{key}{counter}={dev} ")
+            counter+=1
+    print(strings)
+
+
+def handle_usb_children(usb_device, args):
+    """
+    similiar to showtree
+    """
+
+    if len(usb_device.children) != 0:
+        for child_dev in usb_device.children:
+            env_pair_strings(child_dev, args)
+
+
+def fix_label(label : str):
+    return str(label).replace("-", "_").upper()
+
+
+def match_env_with_node(usb_device, args, env : str):
+    """
+    Matches env labels from yml with nodes
+    """
+    # Iterate through the devices associated with this USB device
+    for device in usb_device.devices:
+        # Filter out /dev/bus/usb devices if needed
+        if (
+            not args.show_devusb
+            and device.devname is not None
+            and device.devname.startswith("/dev/bus/usb")
+        ):
```

Comment:
The approach you've taken is to copy show and refactor to support -x - however this leaves lots of duplication such as above - this will make it more difficult to add features later.

I wonder if there is another approach? For example modifying the prints in show to support printing just env's - or perhaps by modifying show to support a formatter callback that's passed in - or something else?